src/test/ui/traits/trait-bounds-not-on-bare-trait.stderr

```diff
@@ -2,9 +2,13 @@ warning: trait objects without an explicit `dyn` are deprecated
   --> $DIR/trait-bounds-not-on-bare-trait.rs:7:12
    |
 LL | fn foo(_x: Foo + Send) {
-   |            ^^^^^^^^^^ help: use `dyn`: `dyn Foo + Send`
+   |            ^^^^^^^^^^ trait object without `dyn`
```

Comment:
This is the kind of case where I think the original output is better - we're pointing to the entire type, so having it inline seems just as clear.

It might be worth it to try to detect this case when constructing `BuiltinLintDiagnostics::BareTraitObject`, and only use the verbose format when the target type is nested within another (e.g. `&TraitName` or `Box<TraitName>`).